ConsoleAppForTesting/MultiThreadingHomeWork.cs

```diff
@@ -0,0 +1,99 @@
+            //    Console.WriteLine("Thread in new thread # {0} now start working, locker = {1} ", Thread.CurrentThread.ManagedThreadId, xLocker);
+            //    //Thread.Sleep(400);
+            //    lock (locker)
+            //    {
+            //        Thread.Sleep(500);
+            //        Console.WriteLine("Second Locker in {0}", Thread.CurrentThread.ManagedThreadId);
+            //        xLocker++;
+            //        Console.WriteLine("Thread in new thread # {0} was stopped, locker = {1}",
+            //            Thread.CurrentThread.ManagedThreadId, xLocker);
+            //    }
+
+            //    myMutex.WaitOne();
+            //    Console.WriteLine("Thread # {0} in mutex code!", Thread.CurrentThread.ManagedThreadId);
+            //    Thread.Sleep(200);
+            //    Console.WriteLine("Thread # {0} out from mutex code!", Thread.CurrentThread.ManagedThreadId);
+            //    myMutex.ReleaseMutex();
+
+            //});
+
+            //var MyThread2 = new Thread(() =>
+            //{
+            //    Console.WriteLine("Thread in new thread # {0} now start working, locker = {1} ", Thread.CurrentThread.ManagedThreadId, xLocker);
+            //    //Thread.Sleep(400);
+            //    lock (locker)
+            //    {
+            //        Console.WriteLine("Second Locker in {0}", Thread.CurrentThread.ManagedThreadId);
+            //        xLocker++;
+            //        Thread.Sleep(500);
+            //        Console.WriteLine("Thread in new thread # {0} was stopped, locker = {1}",
+            //            Thread.CurrentThread.ManagedThreadId, xLocker);
+            //    }
+
+            //    myMutex.WaitOne();
+            //    Console.WriteLine("Thread # {0} in mutex code!", Thread.CurrentThread.ManagedThreadId);
+            //    Thread.Sleep(200);
+            //    Console.WriteLine("Thread # {0} out from mutex code!", Thread.CurrentThread.ManagedThreadId);
+            //    myMutex.ReleaseMutex();
+            //});
+
+            //MyThread2.Start();
+            //myThread.Start();
+            //lock (locker)
+            //{
+            //    Console.WriteLine("General Locker in {0}", Thread.CurrentThread.ManagedThreadId);
+            //    Thread.Sleep(1000);
+            //    xLocker--;
+            //    Console.WriteLine("General Locker gert left from {0}", Thread.CurrentThread.ManagedThreadId);
+            //}
+
+            //myMutex.WaitOne();
+            //Console.WriteLine("Thread # {0} in mutex code!", Thread.CurrentThread.ManagedThreadId);
+            //Thread.Sleep(200);
+            //Console.WriteLine("Thread # {0} out from mutex code!", Thread.CurrentThread.ManagedThreadId);
+            //myMutex.ReleaseMutex();
+
+            //Console.WriteLine("Thread # {0} was stopped, locker = {1}", Thread.CurrentThread.ManagedThreadId, xLocker);
+
+            // Creates 5 philosophers.
+            var philosophersDinner = new PhilosophersCafe(5,3, 20, 100);
```

Comment:
It's not clear what these numbers mean before going to the constructor itself. Notice to add more readability to your code here, e.g. adding the param names like new PhilosophersCafe(countOfPhilosophers = 5, countOfLunchIntake = 3,  timeForLunch = 20, timeForThink = 100)